src/python/WMCore/Services/WMStats/WMStatsWriter.py

```diff
@@ -77,6 +78,27 @@ def uploadData(self, docs):
             self.couchDB.queue(doc)
         return self.couchDB.commit(returndocs=True)
 
+    def bulkUpdateData(self, docs, existingDocs):
+        """
+        :param docs: docs to insert or update
+        :param existingDocs: docs existing in current
+        :return:
+        """
+        if isinstance(docs, dict):
+            docs = [docs]
+        for doc in docs:
+            if doc['_id'] in existingDocs:
+                revList = existingDocs[doc['_id']].split('-')
```

Comment:
I must be missing something. When we want to update a document in couch, don't we post the data we want (including the current revision number) and couch updates the document and bump the revision number?

Reply:
Yes it does but with new_edits=False it overwrites it doesn't do that. And that is needed for the bulk update otherwise we have update document one by one. Only highest revision number is showing when document is retrieved without revision number. That is the reason code gets the current revision number and increase by one.